ctapipe/instrument/subarray.py

```diff
@@ -458,66 +463,75 @@ def to_hdf(self, output_path, overwrite=False):
 
     @classmethod
     def from_hdf(cls, path):
-        layout = Table.read(path, path="/configuration/instrument/subarray/layout")
-
-        optics_table = Table.read(
-            path, path="/configuration/instrument/telescope/optics"
-        )
+        layout = QTable.read(path, path="/configuration/instrument/subarray/layout")
 
         cameras = {}
-        for name in set(layout["camera_type"]):
+
+        # backwards compatibility for older tables, index is the name
+        if "camera_index" not in layout.colnames:
+            layout["camera_index"] = layout["camera_type"]
+
+        for idx in sorted(set(layout["camera_index"])):
```

Comment:
Is the `sorted` necessary? Everything is addresed by the index anyway

Reply:
Index can be the camera name if in "backwards" compatibility mode.

Reply:
right

Reply:
copy paste ;)